src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2136,6 +2194,46 @@ private void emitProperty(
     }
 
     /**
+     * Closure Compiler does not support tuple types, and thus cannot express the proper type for
+     * "Map<K, V>.entries()", which is an IterableIterator of a [K, V] tuple.
```

Comment:
Not clear from the comment why the wider type `(K|V)[]` is not used. It is simply that [K, V] is 'more correct' or  destructuring is not accepted by TS.

Reply:
Expanded the comment.